src/main/java/requestManagement/fleetManager/storage/HostDAOFileSystem.java

```
@@ -0,0 +1,157 @@
+                    if (dataFromFile.get(i).contains(host.getIpv4())) {
+                        dataFromFile.remove(i);
+                    }
+                }
+                writeToFile();
+            }
+        }
+        catch (ArrayIndexOutOfBoundsException e){
+
+        }
+    }
+
+    @Override
+    public void enableHost(Host host) {
+        readFile();
+        try {
+            if (contains(host.getIpv4())) {
+                for (int i = 0; i < dataFromFile.size(); i++) {
+                    if (dataFromFile.get(i).contains(host.getIpv4())) {
+                        dataFromFile.get(i).set(3, "active");
+                        host.setState("active");
+                    }
+                }
+                writeToFile();
+            }
+        }
+        catch (ArrayIndexOutOfBoundsException e){
+
+        }
+    }
+
+    @Override
+    public void disableHost(Host host) {
+        readFile();
+        try {
+            if (contains(host.getIpv4())) {
+                for (int i = 0; i < dataFromFile.size(); i++) {
+                    if (dataFromFile.get(i).contains(host.getIpv4())) {
+                        dataFromFile.get(i).set(3, "inactive");
+                        host.setState("inactive");
+                    }
+                }
+                writeToFile();
+            }
+        }
+        catch (ArrayIndexOutOfBoundsException e){
+
+        }
+    }
+
+    @Override
+    public List<Host> getHosts() {
+        readFile();
+        List<Host> hosts=new ArrayList<Host>();
+        Host tempHost;
+        boolean isPublic=false;
+        try {
+            for (List<String> hostData : dataFromFile) {
+                if (hostData.get(4) == "public")
+                    isPublic = true;
+                tempHost = new Host(hostData.get(3), hostData.get(0), hostData.get(1), hostData.get(2), isPublic);
+                hosts.add(tempHost);
+            }
+        }
+        catch (ArrayIndexOutOfBoundsException e){
+
```

Empty catch block, what's the point of catching it if you're doing nothing? You don't want to swallow errors

---

Done

---

+1